server/api/middlewares/validateClientSecurity.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response, NextFunction } from 'express';
+import { validateS2sToken } from '../services/s2sTokenValidation.service';
+import { validateUserToken } from '../services/userTokenValidation.service';
+import * as process from 'process';
+
+export async function validateIncomingRequest(req: Request, res: Response, next: NextFunction) {
+    let clientServiceName;
+    const urlS2s = process.env.S2S_TOKEN_URL;
+    let rawS2sToken: any = req.headers.serviceauthorization;
```

Comment:
Is any necessary here, or will it always be a string? (In which case, please change the typing)